src/components/AddMovie.jsx

```diff
@@ -1 +1,190 @@
-// implement AddMovie component here
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class AddMovie extends Component {
+  constructor(props) {
+    super(props);
+
+    this.onClick = this.onClick.bind(this);
+    this.resetMovie = this.resetMovie.bind(this);
+    this.inputMovieTitle = this.inputMovieTitle.bind(this);
+    this.inputMovieSubtitle = this.inputMovieSubtitle.bind(this);
+    this.inputMovieImage = this.inputMovieImage.bind(this);
+    this.inputMovieStoryline = this.inputMovieStoryline.bind(this);
+    this.inputMovieRating = this.inputMovieRating.bind(this);
+    this.inputMovieGenre = this.inputMovieGenre.bind(this);
+
+    this.state = {
+      subtitle: '',
+      title: '',
+      imagePath: '',
+      storyline: '',
+      rating: 0,
+      genre: 'action',
+    };
+  }
+
+  onClick({ target }) {
+    const { name, value } = target;
```

Comment:
Bom uso do destructuring, teve algum motivo de não ter descontruído tudo em apenas uma linha?
```suggestion
  onClick({ target:  { name, value } }) {
```